internal/client/client.go

```diff
@@ -50,6 +51,9 @@ func New(infoURL, username, password, authToken string, timeout time.Duration, u
 		if res.StatusCode == http.StatusUnauthorized {
 			return nil, ErrNotAuthorized
 		}
+		if res.StatusCode == http.StatusTooManyRequests {
```

Comment:
I think four cases is the point where I would replace the if-blocks with a switch.